extendr-api/src/iter.rs

```diff
@@ -55,22 +55,18 @@ impl StrIter {
     }
 }
 
-// Get a string reference from a CHARSXP
-fn str_from_strsxp<'a>(sexp: SEXP, index: isize) -> &'a str {
+// Get a string reference from a `CHARSXP`
+pub(crate) fn str_from_strsxp<'a>(sexp: SEXP, index: usize) -> &'a str {
     single_threaded(|| unsafe {
-        if index < 0 || index >= Rf_xlength(sexp) {
+        let charsxp = STRING_ELT(sexp, index as _);
+        if charsxp == R_NilValue {
+            panic!("out of bound error")
```

Comment:
```suggestion
            panic!("Index out of bounds error")
```

Reply:
Thanks! I'm going to remove this part, because it is stupid to have a panic here. I just need to think a bit clearer.. I'll ping when I've restructured it.